src/gateway/router/routes/contractsRoute.ts

```diff
@@ -6,63 +6,58 @@ const MAX_CONTRACTS_PER_PAGE = 100;
-                 max(i.block_height)                                             AS last_interaction_height,
-                 count(*) OVER ()                                                AS total
-          FROM contracts c
-                   LEFT JOIN interactions i
-                             ON c.contract_id = i.contract_id
-                   LEFT JOIN contracts_src s
-                             ON c.src_tx_id = s.src_tx_id
-          WHERE c.contract_id != ''
-            AND c.type != 'error' 
-                ${contractType ? ' AND c.type = ?' : ''} 
-                ${sourceType ? ' AND s.src_content_type = ?' : ''}
-                ${testnet ? ' AND c.testnet IS NOT NULL' : ''}
-          GROUP BY c.contract_id, c.owner, c.type, c.pst_ticker, c.pst_name, s.src_content_type, s.src_wasm_lang
-          ORDER BY last_interaction_height DESC NULLS LAST, interactions DESC NULLS LAST ${
-            parsedPage ? ' LIMIT ? OFFSET ?' : ''
-          };
+        with contract as (select 'contract'      AS contract_or_interaction,
+                                 contract_id     AS contract,
+                                 ''              AS interaction_id,
+                                 owner           AS owner,
+                                 type            AS contract_type,
+                                 ''              AS function,
+                                 block_height    AS block_height,
+                                 deployment_type AS source
+                          from contracts
+                          where contract_id != ''
+                            AND type != 'error' 
+        ${testnet ? ' AND c.testnet IS NOT NULL' : ''}
+                          order by block_height desc 
+                              ${ contractLimit ? ' LIMIT ? ' : '' }),
+            
+            interaction as (select 'interaction'  AS contract_or_interaction,
+                                  contract_id    AS contract,
+                                  interaction_id AS interaction_id,
+                                  owner          AS owner,
+                                  ''             AS contract_type,
+                                  function       AS function,
+                                  block_height   AS block_height,
+                                  source         AS source
+                                from interactions
+                            where contract_id != ''
+          ${testnet ? ' AND testnet IS NOT NULL' : ''}
+                                order by sort_key desc 
+                                ${ interactionLimit ? ' LIMIT ?' : '' })
+                                select * from contract
+                                union all
+                                select * from interaction;
       `,
       bindings
     );
 
-    const total = result?.rows?.length > 0 ? parseInt(result.rows[0].total) : 0;
     ctx.body = {
       paging: {
```

Comment:
i don't know if `paging` name is the correct one here

Reply:
yep, `paging` is misleading, changed to `summary`, but I'm open to other suggestions